libdevice/cmake/modules/SYCLLibdevice.cmake

```diff
@@ -29,152 +20,138 @@ set(compile_opts
+BuildSYCLDeviceLib(Path ${obj_binary_dir}/libsycl-fallback-cmath-spir64_fpga.bc Target spir64_fpga-unknown-unknown-sycldevice Source fallback-cmath.cpp Depends device_math.h device.h clang)
+
+set(devicelib-wrapper-cmath-fp64-spir64 ${obj_binary_dir}/libsycl-cmath-fp64-spir64.bc)
+BuildSYCLDeviceLib(Path ${devicelib-wrapper-cmath-fp64-spir64} Target spir64-unknown-unknown-sycldevice Source cmath_wrapper_fp64.cpp Depends device_math.h device.h clang)
+BuildSYCLDeviceLib(Path ${obj_binary_dir}/libsycl-fallback-cmath-fp64-spir64.bc Target spir64-unknown-unknown-sycldevice Source fallback-cmath-fp64.cpp Depends device_math.h device.h clang)
+
+set(devicelib-wrapper-cmath-fp64-spir64_x86_64 ${obj_binary_dir}/libsycl-cmath-fp64-spir64_x86_64.bc)
+BuildSYCLDeviceLib(Path ${devicelib-wrapper-cmath-fp64-spir64_x86_64} Target spir64_x86_64-unknown-unknown-sycldevice Source cmath_wrapper_fp64.cpp Depends device_math.h device.h clang)
+BuildSYCLDeviceLib(Path ${obj_binary_dir}/libsycl-fallback-cmath-fp64-spir64_x86_64.bc Target spir64_x86_64-unknown-unknown-sycldevice Source fallback-cmath-fp64.cpp Depends device_math.h device.h clang)
+
+set(devicelib-wrapper-cmath-fp64-spir64_gen ${obj_binary_dir}/libsycl-cmath-fp64-spir64_gen.bc)
+BuildSYCLDeviceLib(Path ${devicelib-wrapper-cmath-fp64-spir64_gen} Target spir64_gen-unknown-unknown-sycldevice Source cmath_wrapper_fp64.cpp Depends device_math.h device.h clang)
+BuildSYCLDeviceLib(Path ${obj_binary_dir}/libsycl-fallback-cmath-fp64-spir64_gen.bc Target spir64_gen-unknown-unknown-sycldevice Source fallback-cmath-fp64.cpp Depends device_math.h device.h clang)
+
+set(devicelib-wrapper-cmath-fp64-spir64_fpga ${obj_binary_dir}/libsycl-cmath-fp64-spir64_fpga.bc)
+BuildSYCLDeviceLib(Path ${devicelib-wrapper-cmath-fp64-spir64_fpga} Target spir64_fpga-unknown-unknown-sycldevice Source cmath_wrapper_fp64.cpp Depends device_math.h device.h clang)
+BuildSYCLDeviceLib(Path ${obj_binary_dir}/libsycl-fallback-cmath-fp64-spir64_fpga.bc Target spir64_fpga-unknown-unknown-sycldevice Source fallback-cmath-fp64.cpp Depends device_math.h device.h clang)
+
+function(BuildSYCLFallbackDeviceLib)
+  cmake_parse_arguments(DeviceLib "" "Path;Source" "Depends" ${ARGN})
+  add_custom_command(OUTPUT ${DeviceLib_Path}
+                     COMMAND ${clang} -S -fsycl-device-only -fno-sycl-use-bitcode
+                             ${compile_opts}
+                             ${CMAKE_CURRENT_SOURCE_DIR}/${DeviceLib_Source}
+                             -o ${DeviceLib_Path} -fno-sycl-device-lib=all
+                     MAIN_DEPENDENCY ${DeviceLib_Source}
+                     DEPENDS ${DeviceLib_Depends}
+                     VERBATIM)
+endfunction()
+
+BuildSYCLFallbackDeviceLib(Path ${spv_binary_dir}/libsycl-fallback-cassert.spv Source fallback-cassert.cpp Depends wrapper.h device.h clang spirv_vars.h llvm-spirv)
```

Comment:
It looks like we should rename `wrapper.h` to something like `device_assert` to align with the rest of the library.

Reply:
Hi, @bader 
Currently, the wrapper.h only includes declaration of "__devicelib_assert", we may have more C library function declared here, what about renaming it to "device_crt.h" (crt is "C runtime")?
Thanks very much.

Reply:
I thought that device library code separation logic follows C/C++ standard library layout. AFAIK, assert is located in a separate header file: https://en.cppreference.com/w/cpp/header/cassert.

Reply:
Fixed.